src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -210,7 +216,25 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
                     Resources.FormatViewComponent_IViewComponentFactory_ReturnedNull(descriptor.Type.FullName));
             }
 
-            return invoker.InvokeAsync(context);
+#pragma warning disable 0618
+            if (_diagnosticSource.IsEnabled("Microsoft.AspNet.Mvc.BeforeViewComponent"))
+            {
+                _diagnosticSource.Write(
+                    "Microsoft.AspNet.Mvc.BeforeViewComponent",
+                    new { actionDescriptor = _viewContext.ActionDescriptor, viewComponentContext = context });
+            }
+
+            var result = invoker.InvokeAsync(context);
```

Comment:
Should we have done this change in the invoker? That would allow us to get at the result returned from the VC.